app/presenters/two_factor_authentication/selection_presenter.rb

```diff
@@ -30,13 +38,87 @@ def disabled?
       false
     end
 
-    private
+    def login_label(type)
+      case type
+      when 'auth_app'
+        t('two_factor_authentication.login_options.auth_app')
```

Comment:
Considering that this is a base class, I might expect these methods to be abstract and expected to be overridden in the specific presenter classes, e.g.:

```rb
class SelectionPresenter
  def login_label
    raise "Unsupported login method: #{type}"
  end
end

class AuthAppSelectionPresenter < SelectionPresenter
  def login_label
    t('two_factor_authentication.login_options.auth_app')
  end
end
```

Reply:
That's a good point. The new functions should never be called directly (only via `info` and `label`), so it may make sense to move them to be private?

Reply:
> That's a good point. The new functions should never be called directly (only via `info` and `label`), so it may make sense to move them to be private?

Seems reasonable 👍